clang/lib/AST/Interp/ByteCodeExprGen.cpp

```diff
@@ -827,6 +829,9 @@ bool ByteCodeExprGen<Emitter>::VisitArrayInitLoopExpr(
 
 template <class Emitter>
 bool ByteCodeExprGen<Emitter>::VisitOpaqueValueExpr(const OpaqueValueExpr *E) {
+  if (OpaqueExprs.contains(E))
+    return this->emitGetLocal(*classify(E), OpaqueExprs[E], E);
```

Comment:
Can `E` be of  composite type? If so, we need to handle that. If not,  you can just use `classifyPrim` instead.

This code also potentially does two lookups. you should do `auto It = OpaqueExprs.find(E); It != OpaqueExprs.end()) {`

Reply:
Good question, I found this in the test files:

```
`-OpaqueValueExpr {{.*}} 'result_t':'awaitable_frame::result_t' lvalue
   `-MaterializeTemporaryExpr {{.*}} 'result_t':'awaitable_frame::result_t' lvalue
      `-...
```
```
struct awaitable_frame {
  ...
  struct result_t {
    ~result_t();
    bool await_ready() const noexcept;
    void await_suspend(std::coroutine_handle<void>) noexcept;
    void await_resume() const noexcept;
  };
  ...
};
```
Based on this it seems like it can be a composite type, so I'll make sure this case is handled too.